ratis-server-api/src/main/java/org/apache/ratis/server/RaftServer.java

```diff
@@ -168,21 +172,28 @@ class Builder {
     private static Method initNewRaftServerMethod() {
       final String className = RaftServer.class.getPackage().getName() + ".impl.ServerImplUtils";
       final Class<?>[] argClasses = {RaftPeerId.class, RaftGroup.class, RaftStorage.StartupOption.class,
-          StateMachine.Registry.class, RaftProperties.class, Parameters.class};
+          StateMachine.Registry.class, RaftProperties.class, Parameters.class, Thread.UncaughtExceptionHandler.class};
       try {
         final Class<?> clazz = ReflectionUtils.getClassByName(className);
-        return clazz.getMethod("newRaftServer", argClasses);
+        try {
+          return clazz.getMethod("newRaftServer", argClasses);
+        } catch (NoSuchMethodException e) {
+          final Class<?>[] argClassesNoExHandler = {RaftPeerId.class, RaftGroup.class, RaftStorage.StartupOption.class,
+              StateMachine.Registry.class, RaftProperties.class, Parameters.class};
+          return ReflectionUtils.getClassByName(className).getMethod("newRaftServer", argClassesNoExHandler);
```

Comment:
This is an ugly attempt to keep backward compatibility. I totally have no idea how many people have in-house implementations of `RaftServer` and things may break because of my change. PTAL if this is okay and i'll really appreciate it if you could point out a better way :)

Reply:
@szetszwo PTAL if this is okay, thanks a lot!

Reply:
Ok i misunderstood the code here. I initially thought this is to invoke user extensions (i.e. a user may extend `ServerImplUtils.newRaftServer()` to init a `RaftServer` child class), so the `newRaftServer()` signature should be kept the same to have backward compatibility. 

Then after a while I realized `ServerImplUtils` is a `final` class so no extension is expected. I guess this reflection is used just to avoid importing `ratis-server` module (avoid circular dependency). 

Therefore no backward compatibility is needed here.